### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by akshayp7** [akshayp7/playwright-typescript-playwright-test#47] [Announcements]
Dedicated to Nikola Tesla - Inventor of Alternating Current and much more
Changelog:

Changed POM structure and used new locators as per playwright best practices
Changed docker file to install chrome.
Fixed error in import path
Updated to official allure-playwright reporter
Updated to Playwright 1.31.0 .
and much more under the hood ;)

**Comment by AutoQA23** [comment]
Hi @akshayp7 
You have removed Objects? in the latest. What is the reason?

**Reply by akshayp7** [reply]
I have kept locator objects in page file. This is as per Playwright best practice. 

https://playwright.dev/docs/pom

**Comment by nchukwuocha** [comment]
Hi akshap7,

How can I use the test explorer in VS Code to have it recognize individual tests and also be able to use an environment from your boilerplate? If I use your boilerplate I can run tests in the command line, but I cannot pick a specific test with the "npx cross-env ENV=qa npm run test:single" command. But then, if I use the basic playwright template that comes with when you install playwright, I can select different tests I want to run but I cannot select an environment.
![testfix4](https://github.com/akshayp7/playwright-typescript-playwright-test/assets/134742530/48d6c472-02c2-4fb7-9e7a-235726a0e06a)

**Reply by akshayp7** [reply]
Hi . Thanks for your feedback. I think that is because cross-env is a different library. Let me check and get back to you.

**Reply by nchukwuocha** [reply]
Hi,

Thanks for reply. I guess the main question would be what I would need to add to the playwright.config.ts for it to appear without disabling the cross env. but thanks again for checking.

**Reply by AutoQA23** [reply]
> Hi akshap7,
> 
> How can I use the test explorer in VS Code to have it recognize individual tests and also be able to use an environment from your boilerplate? If I use your boilerplate I can run tests in the command line, but I cannot pick a specific test with the "npx cross-env ENV=qa npm run test:single" command. But then, if I use the basic playwright template that comes with when you install playwright, I can select different tests I want to run but I cannot select an environment. ![testfix4](https://user-images.githubusercontent.com/134742530/241704274-48d6c472-02c2-4fb7-9e7a-235726a0e06a.gif)

I am after same situation haven't been successful with a solution yet.

**Reply by akshayp7** [reply]
Hi , I am thinking of a way where we can remove this cross-env dependency and try to provide ENV variables. After that I  think it might solve the issue just a guess. I am also looking for solution. It might take some time but thank you for brining this up

**Reply by akshayp7** [reply]
Hi I have made changes in code instead of using cross-env now I am using process-env. Can you please check this issue once and let me know if it's fixed

**Reply by AutoQA23** [reply]
@akshayp7 The changes you made allows to use trace viewer not  Test explorer. I find test explorer more useful as it had debugging step by step along with breakpoints which is nicely integrated with IDE. I think trace viewer is useful when you are more after performance and network analysis. It would be great if we can use Test Explorer in your framework. 
![image](https://github.com/akshayp7/playwright-typescript-playwright-test/assets/118112890/c3ab6a95-4174-4305-8e7d-8a95f9b03b2c)

**Reply by akshayp7** [reply]
Let me look into it. Please let me know if you find something as well

**Comment by akshayp7** [comment]
Hi I tried to implement dotenv solution but the problem is we can't pass qa or dev environment variables from command line we have to pass it via env config file.

**Reply by AutoQA23** [reply]
I am not sure the need of passing env from command line?

**Reply by akshayp7** [reply]
It's dynamic sometimes you need to pass qa or dev environment for example if you are running on ci/cd. What if you want to run same set of test cases on dev and qa. Commiting the changes and pushing code everytime would be cumbersome so we can pass qa or dev from command line while executing tests

**Reply by AutoQA23** [reply]
Right, might also depends on the project needs unless if you need to run the tests every time on other environments.
